cli/flox/src/commands/upload.rs

```diff
@@ -0,0 +1,134 @@
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::providers::publish::{BinaryCache, NixCopyCache};
+use tracing::instrument;
+use url::Url;
+
+use crate::config::Config;
+use crate::subcommand_metric;
+use crate::utils::dialog::{Dialog, Spinner};
+use crate::utils::message;
+
+#[derive(Bpaf, Clone)]
+pub struct Upload {
+    #[bpaf(external(cache_args))]
+    cache: CacheArgs,
+
+    #[bpaf(external(upload_store_path))]
+    store_path: UploadStorePath,
+}
+
+#[derive(Debug, Bpaf, Clone)]
+struct CacheArgs {
+    #[bpaf(long("cache"))]
+    url: Url,
+
+    #[bpaf(long("signing-key"))]
+    key_file: PathBuf,
+}
+
+#[derive(Debug, Bpaf, Clone)]
+struct UploadStorePath {
+    /// The store path to upload.
+    #[bpaf(positional("store-path"))]
+    store_path: PathBuf,
+}
```

Comment:
nit (nb): technically there is no need to break these into subparsers, especially without anyo documentation attached to the groups.